examples/wallet/app/utils/nodeConnection.js

```diff
@@ -25,74 +25,127 @@ const GRPC_URL = `${NODE_URL}:${GRPC_PORT}`;
-  return axios
-    .get(`${BASE_REST_URL}/account/${identifier}`)
-    .then(({ data: { value, counter, delegation } }) => ({
-      balance: value,
-      counter,
-      delegation: delegation.pools.reduce(
-        (acc, [poolId, amount]) =>
-          Object.assign(acc, { [poolId]: { parts: amount } }),
-        {}
-      )
-    }))
-    .catch(error => {
-      // When an account is imported with BIP39, the account may not
-      // exist in the blockchain. For this DEMO we hardcode the account
-      // information when an HTTP 404 occurs because we interpret that
-      // the account exists. If any type of error occurs, an exception
-      // is thrown to be handled in the upper layers.
-      if (error && error.response && error.response.status === 404) {
-        const balance: Balance = 0;
-        const counter: Counter = 0;
-        const delegation: Delegation = { '00000x': { parts: 1 } };
-        return {
-          balance,
-          counter,
-          delegation
-        };
-      }
-      throw error;
-    });
-}
+export type BroadcastTransaction = $PropertyType<
+  $Call<typeof nodeConnectionBuilder>,
+  'broadcastTransaction'
+>;
 
-const flattenInputOrOutput = (
-  arr: Array<{ amount: number, address: { id: Address } }>
-) =>
-  arr.map(({ address, amount }) => ({
-    amount: Number(amount),
-    address: address.id
-  }));
+export default function nodeConnectionBuilder({
+  injectedAxios,
+  injectedGrpcClient
+}: {
+  injectedAxios?: any,
+  injectedGrpcClient?: any
+} = {}) {
+  const axiosInstance = ((injectedAxios: any): typeof axios) || axios;
+  const grpcClientInstance =
+    (injectedGrpcClient: typeof grpcClient) || grpcClient;
+  const getAccountState = function getAccountState(
+    identifier: Identifier
+  ): Promise<AccountState> {
+    return axiosInstance
+      .get(`${BASE_REST_URL}/account/${identifier}`)
+      .then(({ data: { value, counter, delegation } }) => ({
+        balance: value,
+        counter,
+        delegation: delegation.pools.reduce(
+          (acc, [poolId, amount]) =>
+            Object.assign(acc, { [poolId]: { parts: amount } }),
+          {}
+        )
+      }))
+      .catch(error => {
+        // When an account is imported with BIP39, the account may not
+        // exist in the blockchain. For this DEMO we hardcode the account
+        // information when an HTTP 404 occurs because we interpret that
+        // the account exists. If any type of error occurs, an exception
+        // is thrown to be handled in the upper layers.
+        if (error && error.response && error.response.status === 404) {
+          const balance: Balance = 0;
+          const counter: Counter = 0;
+          const delegation: Delegation = { '00000x': { parts: 1 } };
+          return {
+            balance,
+            counter,
+            delegation
+          };
+        }
+        throw error;
+      });
+  };
```

Comment:
When the application is run with **yarn dev** or **yarn start**, after removing **node_modules**, **yarn.lock** and **npm install**, it takes a long time to open the instance. Can this builder influence the opening of the wallet?

Reply:
did you try to do the same on a different revision?

Reply:
going from zero to 'app running' involves three stages:

- installing the dependencies: this is done via `yarn install` and takes a lot of time because the actual dependency tree for this and every node proyect is huge. This has been widely memed about: https://www.improgrammer.net/npm-package-delivery/

- building the renderer and main 'processes': this is done via `yarn build`, and consists of transpiling all our source code and assets into a 'bundle' that a browser can actually run. Webpack takes care of this, and no changes were introduces in this PR that might increase transpilation time.

- running the app: this consists of firing up the browser and actually running our (transpiled) code.
Having the builder only adds one syncronous function call to inject the dependencies, with nothing taking up particularly long-running. That's why I'm pretty skeptical of the builder increasing the load time. I'd however be pretty curious to try to replicate any benchmark 🙃.

Reply:
I think It's Ok. worked with node 10.